src/main/java/org/deepsymmetry/beatlink/dbserver/ConnectionManager.java

```diff
@@ -522,6 +540,9 @@ public synchronized void stop() {
         if (isRunning()) {
             running.set(false);
             DeviceFinder.getInstance().removeDeviceAnnouncementListener(announcementListener);
+            for (Thread thread : activeQueryThreads.values()) {
```

Comment:
And finally here is the new code to interrupt any threads that are querying for DB server ports when we are told to shut down.